pandas/plotting/_matplotlib/core.py

```diff
@@ -124,13 +126,34 @@ def __init__(
         table=False,
         layout=None,
         include_bool=False,
+        column: IndexLabel | None = None,
         **kwds,
     ):
 
         import matplotlib.pyplot as plt
 
         self.data = data
-        self.by = by
+        self.by = com.maybe_make_list(by)
+
+        # Assign the rest of columns into self.columns if by is explicitly defined
+        # while column is not, so as to keep the same behaviour with current df.hist
+        # or df.boxplot.
+        if self.by and column is None:
+            self.columns = [
+                col
+                for col in data.columns
+                if col not in self.by and is_numeric_dtype(data[col])
+            ]
+        else:
+            self.columns = com.maybe_make_list(column)
+
+        # When `by` is explicitly assigned, grouped data size will be defined, and
+        # this will determine number of subplots to have, aka `self.nseries`
+        if self.by:
+            if self._kind == "hist":
+                self._grouped_data_size = len(data.groupby(self.by))
+            elif self._kind == "box":
+                self._grouped_data_size = len(self.columns)
```

Comment:
yeah! I moved here. Only that the `self.data` needs to be changed a bit, since `self.data` is updated later, and what we need here is the original grouped data length. Please take a look at the change here and see if it is okay for you?